saleor/discount/interface.py

```diff
@@ -59,10 +59,14 @@ def fetch_variant_rules_info(
         if variant_channel_listing
         else []
     )
+
     rules_info = []
-    for listing_promotion_rule in listings_rules:
+    if listings_rules:
+        listing_promotion_rule = max(
```

Comment:
Is there a reason why max? Maybe some comment here would be useful.

Reply:
Added: 

> Before introducing unique_type on discount models, there were possibility
   to have multiple catalogue discount associated with single line. In such a
   case, we should pick the best discount (with the highest discount amount)

Reply:
@korycins I don't think it is necessary. Just checked and `update_discounted_prices_for_promotion` is currently designed to keep single `VariantChannelListingPromotionRule` relation (while db allows us to keep multiples, we don't use it now)